### DISCUSSION TITLE
Problem in backward

### DISCUSSION THREAD

**Discussion by haoweiz23** [huggingface/diffusers#7494] [Q&A]
### Describe the bug

Hi, I modified the U-Net forward function to use the "wrapped_forward" function. However, I encountered an issue when attempting to backpropagate the loss. I adapted the training code from "examples/train_dreambooth.py". The error log is provided below. It appears to be related to gradient checkpointing, but I couldn't find any useful information to resolve it.
![image](https://github.com/huggingface/diffusers/assets/43101596/198290f1-59dd-42f9-a1b0-b0f6e156f691)

### Reproduction

![image](https://github.com/huggingface/diffusers/assets/43101596/f4db2133-96a3-4dfd-966c-05f4c71ed59d)

### System Info

Copy-and-paste the text below in your GitHub issue and FILL OUT the two last points.

- `diffusers` version: 0.28.0.dev0
- Platform: Linux-5.4.0-148-generic-x86_64-with-glibc2.31
- Python version: 3.10.6
- PyTorch version (GPU?): 1.13.1+cu117 (True)
- Huggingface_hub version: 0.20.2
- Transformers version: 4.39.0
- Accelerate version: 0.25.0
- xFormers version: not installed
- Using GPU in script?: <fill in>
- Using distributed or parallel set-up in script?: <fill in>

**Comment by sayakpaul** [comment]
The error seems to be coming from a piece code that lies outside of the library. So, I suggest you open a discussion for this instead of an issue.

**Comment by tolgacangoz** [comment]
Hi @haoweiz23,
To be honest, I couldn't understand what is going on just to see these two pictures. Could you please share your modified examples/train_dreambooth.py and U-Net files and give details about the intention of your wrapping?
